saleor/order/tests/test_fetch_order_prices.py

```diff
@@ -2637,18 +2696,298 @@ def test_fetch_order_prices_voucher_shipping_percentage(
+
+    undiscounted_shipping_price = order.shipping_price.net.amount
+    base_shipping_price = Decimal(undiscounted_shipping_price - voucher_discount_amount)
+    currency = order.currency
+    undiscounted_subtotal = Decimal(0)
+    lines = order.lines.all()
+    for line in lines:
+        undiscounted_subtotal += line.base_unit_price_amount * line.quantity
+
+    # create manual order discount
+    manual_discount_value = Decimal("1000")
+    manual_discount = order.discounts.create(
+        value_type=DiscountValueType.FIXED,
+        value=manual_discount_value,
+        name="Manual order discount",
+        type=DiscountType.MANUAL,
+        currency=currency,
+    )
+
+    total_with_shipping_voucher_discount = undiscounted_subtotal + base_shipping_price
+    assert manual_discount_value > total_with_shipping_voucher_discount
+    manual_discount_amount = total_with_shipping_voucher_discount
+    subtotal_manual_reward_portion = round(
+        (undiscounted_subtotal / total_with_shipping_voucher_discount)
+        * manual_discount_amount,
+        2,
+    )
+    shipping_manual_reward_portion = round(
+        manual_discount_amount - subtotal_manual_reward_portion, 2
+    )
+
+    expected_subtotal = undiscounted_subtotal - subtotal_manual_reward_portion
+    expected_shipping = base_shipping_price - shipping_manual_reward_portion
+    expected_total = expected_subtotal + expected_shipping
+
+    # when
+    order, lines = calculations.fetch_order_prices_if_expired(
+        order, plugins_manager, None, True
+    )
+
+    # then
+    assert OrderDiscount.objects.count() == 2
+    voucher_discount = order.discounts.filter(type=DiscountType.VOUCHER).first()
+    assert voucher_discount.voucher == voucher
+    assert voucher_discount.voucher_code == code
+    assert (
+        voucher_discount.amount_value
+        == undiscounted_shipping_price - base_shipping_price
+    )
+
+    manual_discount.refresh_from_db()
+    assert manual_discount.amount.amount == total_with_shipping_voucher_discount
+
+    assert order.undiscounted_base_shipping_price_amount == undiscounted_shipping_price
+    assert order.base_shipping_price_amount == base_shipping_price
+    assert order.shipping_price_net_amount == expected_shipping
+    assert order.shipping_price_gross.amount == round(expected_shipping * tax_rate, 2)
+    assert order.subtotal_net_amount == expected_subtotal
+    assert order.subtotal_gross_amount == round(expected_subtotal * tax_rate, 2)
+    assert order.total_net_amount == expected_total
+    assert order.total_gross_amount == round(expected_total * tax_rate, 2)
+    # FIXME: If discounts exceed total value, undiscounted prices are set to 0
+    # assert (
+    #     order.undiscounted_total_net_amount
+    #     == undiscounted_subtotal + undiscounted_shipping_price
+    # )
+    # assert order.undiscounted_total_gross_amount == round(
+    #     (undiscounted_subtotal + undiscounted_shipping_price) * tax_rate, 2
+    # )
```

Comment:
What about that?

Reply:
Here is tax: https://linear.app/saleor/issue/SHOPX-1390

Reply:
Could you add a task number in the comment?

Reply:
done